### ISSUE TITLE
Can't jump to definition on first char of word

### ISSUE THREAD

**Issue by maxzhao** [zigtools/zls#891]
### Zig Version

x86_64-0.11.0-dev.1240+24b4e643f

### Zig Language Server Version

2717b0fba19daba897f86dd6de4b8b590bfee516

### Steps to Reproduce

I use Nvim + Lsp,  When I locate on the first char of the fn name, then applied  go to definition action, it does't jump to fn source.  

### Expected Behavior

jump to right definition

### Actual Behavior

nothing happen

**Comment by InKryption** [comment]
I'll attest this also seems to affect single-char identifiers (unable to jump to definition of a declaration of name `i`).

**Comment by acristoffers** [comment]
From the small test I did, it looks like the code was written with VSCode in mind, in the sense that it considers the cursor to be inside a symbol only when it makes sense for a line-cursor. But in block-cursors, usually employed in terminal editors, it makes sense visually to consider the cursor to be in the word when it is right before it.

Those two cursors are on the same position in the file:

![image](https://user-images.githubusercontent.com/2191364/211194235-781d347c-cd25-4c26-aaf7-4e21eb75f7dd.png)

So for the block-cursor, it is pretty much in the symbol, while for the line-cursor it may be debatable.

This is also the reason why single-char identifiers may not work in VSCode. For me in Emacs it works only if I have the cursor after the identifier.

I tried to look at the code, but I'm not familiar enough to reason about it. It seems that the function `identifierFromPosition` (https://github.com/zigtools/zls/blob/master/src/Server.zig#L826) needs to be reworked, but then again the logic there seems right and it should get the symbol corretly. I can't see why the "look-forward" part of the code is not selecting the identifier (I expected the first loop to not run, but the second one should select the identifier). I'll see if I can find a way to debug it.

**Comment by acristoffers** [comment]
I found out that the function `identifierFromPosition` is not really involved in the process. The actual culprid is `getPositionContext`, which actually does follow the line vs block logic I described in the last comment. When the cursor is right after the first char, it correctly returns `analysis.PositionContext{ .field_access }`, but returns `analysis.PositionContext{ .nothing }` when it is before the first char, as it cannot see it.

In the line https://github.com/zigtools/zls/blob/master/src/analysis.zig#L1532, we can just change

```zig
const line_loc = offsets.lineLocUntilIndex(text, doc_index);
```

to

```zig
const line_loc = offsets.lineLocUntilIndex(text, doc_index + 1);
```

to fix the issue. It then reports the right symbol no matter the type of cursor and does not seems to affect the end of symbol, that is, having the cursor before and after the last symbol works as expected.

But I won't oppen a pull request as I'm not confident this is the best approach, I'm not familiar with the code, so I'll wait for a contributor to at least take a look fist.
